export/securedrop_export/print/service.py

```diff
@@ -113,143 +97,30 @@ def printer_test(self) -> Status:
         # a success status here
         return Status.PRINT_TEST_PAGE_SUCCESS
 
-    def _wait_for_print(self):
-        """
-        Use lpstat to ensure the job was fully transfered to the printer
-        Return True if print was successful, otherwise throw ExportException.
-        Currently, the handler `handler` is defined in `exceptions.py`.
-        """
-        signal.signal(signal.SIGALRM, handler)
-        signal.alarm(self.printer_wait_timeout)
-        printer_idle_string = f"printer {self.printer_name} is idle"
-        while True:
-            try:
-                logger.info(f"Running lpstat waiting for printer {self.printer_name}")
-                output = subprocess.check_output(["lpstat", "-p", self.printer_name])
-                if printer_idle_string in output.decode("utf-8"):
-                    logger.info("Print completed")
-                    return True
-                else:
-                    time.sleep(5)
-            except subprocess.CalledProcessError:
-                raise ExportException(sdstatus=Status.ERROR_PRINT)
-            except TimeoutException:
-                logger.error(f"Timeout waiting for printer {self.printer_name}")
-                raise ExportException(sdstatus=Status.ERROR_PRINT)
-        return True
-
     def _check_printer_setup(self) -> None:
         """
         Check printer setup.
         Raise ExportException if supported setup is not found.
         """
         try:
-            logger.info("Searching for printer")
-            output = subprocess.check_output(["sudo", "lpinfo", "-v"])
-            printers = [x for x in output.decode("utf-8").split() if "usb://" in x]
-            if not printers:
-                logger.info("No usb printers connected")
-                raise ExportException(sdstatus=Status.ERROR_PRINTER_NOT_FOUND)
-
-            supported_printers = [
-                p for p in printers if any(sub in p for sub in self.SUPPORTED_PRINTERS)
-            ]
-            if not supported_printers:
-                logger.info(f"{printers} are unsupported printers")
-                raise ExportException(sdstatus=Status.ERROR_PRINTER_NOT_SUPPORTED)
-
-            if len(supported_printers) > 1:
-                logger.info("Too many usb printers connected")
-                raise ExportException(sdstatus=Status.ERROR_MULTIPLE_PRINTERS_FOUND)
+            logger.info("Searching for IPP printers using IPP-USB:")
+            ipp_usb_output = subprocess.check_output(
+                ["sudo", "ipp-usb", "check"], universal_newlines=True
```

Comment:
I would like us to avoid sudo if possible, and I think we can trivially do that here.

Despite being in sbin, we can run `/usr/sbin/ipp-usb status` without needing root. I dug into the ipp-usb code (it's golang and not C like I had feared!) and it turns out it's just connecting over a unix socket, the status command is functionally identical to:

`curl --unix-socket /var/ipp-usb/ctrl http://localhost/status`

But that command started failing when I had no devices connected, and I realized that ipp-usb removes that socket if there are no devices.

So we can simplify this down to `Path("/var/ipp-usb/ctrl").exists()`. I think :) 

I would still suggest we run `ipp-usb status` and send that output to the debug log though.

Reply:
> So we can simplify this down to `Path("/var/ipp-usb/ctrl").exists()`. I think :)

I've asked at https://github.com/OpenPrinting/ipp-usb/discussions/111

Reply:
The response was:
> I think, the most reliable way is to use Avahi to tell, if there is at least one _ipp._tcp or _uscan._tcp service, connected to the local host.

I didn't look into how to do that yet, but if it's not trivial I'm okay leaving that as a follow-up, just add a TODO for it.

Reply:
Yep! I would still prefer to have some comment explaining that it's not a perfect check but good enough for our purposes.

Reply:
👍🏾